File: scala/fizzbuzz/fizzbuzz.scala
object fizzbuzz {
  def fizzbuzz(i: Int) = (i % 3, i % 5) match {
    case (0, 0) => "FizzBuzz"
    case (0, _) => "Fizz"
    case (_, 0) => "Buzz"
    case _      => i
  }
  def main(args: Array[String]) :Unit = {
    (1 to 100).map(fizzbuzz).foreach(println)
  }
}

File: haskell/fizzbuzz/fizzbuzz.hs
fizzBuzz :: Int -> String

fizzBuzz n | n `mod` 15 == 0 = "FizzBuzz"
           | n `mod`  3 == 0 = "Fizz"
           | n `mod`  5 == 0 = "Buzz"
           | otherwise       = show n

main = do
    print $ map fizzBuzz [1..100]